api/controllers/users.js

```diff
@@ -1,24 +1,61 @@
 const User = require('../models/user');
+const cloudinary = require('cloudinary').v2;
+require('dotenv').config();
+
+cloudinary.config({
+  cloud_name: 'de2y4osyf',
+  api_key: process.env.API_KEY,
+  api_secret: process.env.API_SECRET,
+});
 
 const UsersController = {
   Create: (req, res) => {
-    const user = new User({
-      email: req.body.email,
-      password: req.body.password,
-      display_name: req.body.display_name,
-    });
-
+    const seed = Math.round(Math.random() * 100);
+    const avatarUrl = `https://avatars.dicebear.com/api/personas/${seed}.svg`;
     if (req.file) {
-      user.image = req.file.buffer;
-    }
+      cloudinary.uploader
+        .upload_stream(
+          { resource_type: 'raw', format: 'jpg' },
+          (error, result) => {
+            if (error) {
+              console.error(error);
+              res.status(400).json({ message: 'Internal server error' });
+            } else {
+              const image = result.url;
+              const user = new User({
+                email: req.body.email,
+                password: req.body.password,
+                image,
+                display_name: req.body.display_name,
+              });
 
-    user.save((err) => {
-      if (err) {
-        res.status(400).json({ message: 'Bad request' });
-      } else {
-        res.status(201).json({ message: 'OK' });
-      }
-    });
+              user.save((err) => {
+                if (err) {
+                  res.status(400).json({ message: 'Bad request' });
+                } else {
+                  res.status(201).json({ message: 'OK' });
+                }
+              });
+            }
+          }
+        )
+        .end(req.file.buffer);
+    } else {
+      const user = new User({
+        email: req.body.email,
```

Comment:
This method is well written and, for its length, very easy to follow. I can follow the flow of the code without being familiar with cloudinary at all. However, as this method is now 50 lines, it may be beneficial for modularisation purposes to split this method into helper methods that will clarify the purpose of each segment of code. A good one may be "Upload image".